aptos-move/framework/aptos-framework/sources/Coin.move

```diff
@@ -18,41 +18,51 @@ module AptosFramework::Coin {
     const ENO_MINT_CAPABILITY: u64 = 7;
     const EDESTRUCTION_OF_NONZERO_TOKEN: u64 = 8;
 
-    // Core data structures
+    /// Core data structures
 
-    // Represents a set amount of coin
+    /// Main structure representing a coin/token in an account's custody.
     struct Coin<phantom CoinType> has store {
+        /// Amount of coin this address has.
         value: u64,
     }
 
-    // Represents ownership of coin
+    /// A holder of a specific coin type. This is a wrapper that allows
+    /// adding more data such as events and can be thought of as a "wallet".
     struct CoinStore<phantom CoinType> has key {
         coin: Coin<CoinType>,
         deposit_events: EventHandle<DepositEvent>,
         withdraw_events: EventHandle<WithdrawEvent>,
     }
 
-    // Represents resources associated with a coin
+    /// Information about a specific coin type declared on the coin creator.
     struct CoinInfo<phantom CoinType> has key {
         name: ASCII::String,
-        scaling_factor: u64,
+        /// Symbol of the coin, usually a shorter version of the name.
+        symbol: ASCII::String,
+        /// Number of decimals used to get its user representation/
+        decimals: u64,
+        // Amount of this coin type in existence.
         supply: Option<u64>,
     }
 
-    // Set of data sent to the event stream during a receive
+    /// Event emitted when some amount of a coin is deposited into an account.
     struct DepositEvent has drop, store {
         amount: u64,
     }
 
-    // Set of data sent to the event stream during a withdrawal
+    /// Event emitted when some amount of a coin is withdrawn from an account.
     struct WithdrawEvent has drop, store {
         amount: u64,
     }
 
-    /// Capability required to mint coins.
+    /// Capability required to mint the coin. This is usually given to either the
+    /// account that declared the coin type module or a custom wrapper module that
```

Comment:
how about just keep "Capability required to mint the coin" in case the comment may diverge from how we will use it in future?

Reply:
Updated